casanovo/casanovo.py

```diff
@@ -146,91 +145,151 @@ def sequence(
-    type=click.Path(dir_okay=False),
+    "--enzyme",
+    help="Enzyme for in silico digestion, \
+    See pyteomics.parser.expasy_rules for valid enzymes",
+    type=click.Choice(
+        [
+            "arg-c",
+            "asp-n",
+            "bnps-skatole",
+            "caspase 1",
+            "caspase 2",
+            "caspase 3",
+            "caspase 4",
+            "caspase 5",
+            "caspase 6",
+            "caspase 7",
+            "caspase 8",
+            "caspase 9",
+            "caspase 10",
+            "chymotrypsin high specificity",
+            "chymotrypsin low specificity",
+            "clostripain",
+            "cnbr",
+            "enterokinase",
+            "factor xa",
+            "formic acid",
+            "glutamyl endopeptidase",
+            "granzyme b",
+            "hydroxylamine",
+            "iodosobenzoic acid",
+            "lysc",
+            "ntcb",
+            "pepsin ph1.3",
+            "pepsin ph2.0",
+            "proline endopeptidase",
+            "proteinase k",
+            "staphylococcal peptidase i",
+            "thermolysin",
+            "thrombin",
+            "trypsin",
+            "trypsin_exception",
+        ]
+    ),
+    default="trypsin",
 )
 @click.option(
-    "-v",
-    "--verbosity",
-    help="""
-    Set the verbosity of console logging messages. Log files are
-    always set to 'debug'.
-    """,
+    "--digestion",
+    help="Digestion: full, partial",
     type=click.Choice(
-        ["debug", "info", "warning", "error"],
+        ["full", "partial"],
         case_sensitive=False,
     ),
-    default="info",
+    default="full",
 )
-def annotate(
-    peak_path: str,
-    tide_path: str,
-    output: Optional[str],
-    verbosity: str,
-) -> None:
-    """Annotate a given .mgf with candidates as selected by a Tide search for Casanovo-DB.
-
-    PEAK_PATH must be one MGF file from which to annotate spectra.
-
-    TIDE_PATH must be one directory containing the Tide search results of the <PEAK_PATH> .mgf.
-    This directory must contain tide-search.decoy.txt and tide-search.target.txt
-    """
-    if output is None:
-        output = setup_logging(output, verbosity)
-        logger.info(
-            "Output file not specified. \
-            Annotated MGF will be saved in the same directory \
-            as the input MGF."
-        )
-        output = peak_path.replace(".mgf", "_annotated.mgf")
-    else:
-        output = setup_logging(output, verbosity)
-
-    annotate_mgf(peak_path, tide_path, output)
-
-    logger.info("DONE!")
-
-
-@main.command(cls=_SharedParams)
-@click.argument(
-    "peak_path",
-    required=True,
-    nargs=-1,
-    type=click.Path(exists=True, dir_okay=False),
+@click.option(
+    "--missed_cleavages",
+    help="Number of allowed missed cleavages",
+    type=int,
+    default=0,
+)
+@click.option(
+    "--max_mods",
+    help="Maximum number of modifications per peptide",
+    type=int,
+    default=0,
```

Comment:
In `tide-index`, it seems that the number of modifications can be specified per modification. As in: mod A can be applied max three times per peptide but you can set mod B to only apply max once per peptide. Pyteomics `isoforms` method does not allow this level of granularity and instead just has a variable that specifies the max number of times modifications of any time can occur on a peptide. This CLI option is that number. If the user does not want this restricted, they can just set the number to 999 very easily.

I can change the default value to some unrestricted number when I make sure the modifications themselves can be specified (right now they are static in the code as you have mentioned)

Reply:
Can you loop over `isoforms` and add each modification with a desired maximum count one by one?

Although I don't feel too strongly about it. The key issue is that with the default of 0, if people specify variable modifications, they actually still won't be included, which will be confused. Instead, the default here should be `None` so that all possible peptidoforms are generated when variable modifications are specified, and only if those need to be restricted, this parameter can be changed.